### ISSUE TITLE
[bug]: EventListener not functioning on Android, but works perfectly on iOS.

### ISSUE THREAD

**Issue by RoshanNepal** [WalletConnect/modal-react-native#52]
### Description

I am attempting to trigger a personal sign when the chainId is changed. This logic functions properly on iOS; however, on Android, the event change is not being executed.

```
  React.useEffect(() => {
    const handleChainChanged = (newChainId: string) => {
      setLoading(false)
      if (newChainId === chainId) {
        _personalSign();
      }
    }

    if (isConnected) {
      provider?.events.addListener('chainChanged', handleChainChanged);
    }

    return () => {
      if (isConnected) {
        provider?.events.removeAllListeners();
      }
    }
  }, [isConnected, provider]);

```

### WalletConnect Modal SDK version

1.0.0-rc.3

### Output of `npx react-native info`

System:
    OS: macOS 13.4
    CPU: (10) arm64 Apple M2 Pro
    Memory: 61.50 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.2.0 - /opt/homebrew/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 9.6.6 - /opt/homebrew/bin/npm
    Watchman: 2023.05.22.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.12.1 - /Users/roshan/.rvm/gems/ruby-2.7.6/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.4, iOS 16.4, macOS 13.3, tvOS 16.4, watchOS 9.4
    Android SDK: Not Found
  IDEs:
    Android Studio: 2022.2 AI-222.4459.24.2221.10121639
    Xcode: 14.3.1/14E300c - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.19 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.2.0 => 18.2.0 
    react-native: 0.71.8 => 0.71.8 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

1. Execute switchChain.
2. Add `chainsChanged` event listener 

### Snack, code example, screenshot, or link to a repository

```
React.useEffect(() => {
    const handleChainChanged = (newChainId: string) => {
      if (newChainId === chainId) {
        console.log('execute personal_sign')
      }
    };

    provider.events.addListener('chainChanged', handleChainChanged);

    return () => {
      provider?.events.removeAllListeners();
    };
  }, []);
```

SwitchChain: 

```
export const switchChain = async (provider: any) => {
  try {
    await provider?.request({
      method: "wallet_switchEthereumChain",
      params: [{ chainId: chainIdHex }],
    });
  } catch (error) {
    await provider.request({
      method: "wallet_addEthereumChain",
      params: [
        {
          chainName,
          chainId: chainIdHex,
          nativeCurrency,
          rpcUrls,
          blockExplorerUrls,
        },
      ],
    });
  }
}
```

**Comment by ignaciosantise** [comment]
Oh i think i was able to reproduce the issue. Are you using MetaMask to test? Apparently after calling `wallet_addEthereumWallet` on Android and coming back to the dapp, the chain event doesn't trigger. I'll dig into this and check what's the cause

**Comment by RoshanNepal** [comment]
Hey @ignaciosantise, the problem still exists even after replacing `provider?.events.removeAllListeners();`. Yes, I'm using MetaMask to test.

**Comment by nhatnguyenbp** [comment]
@RoshanNepal maybe same with my issue https://github.com/MetaMask/metamask-mobile/issues/6706

**Comment by ignaciosantise** [comment]
@RoshanNepal @nhatnguyenbp yep, it's the same issue. There's nothing i can do from the SDK side, as we are not receiving `chainChanged` event from MM. 

If you open MM again after being redirected to the dapp, you'll see that the event is now sent, and the chain is updated. There's probably a race condition or something on MM side, and they redirect before sending the `chainChanged` event


https://github.com/WalletConnect/modal-react-native/assets/25931366/13ccf69c-e068-4376-80ab-1047c2a2af89

Im closing this as it's an issue with MM
